cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp

```diff
@@ -462,6 +463,120 @@ sycl::event compute_hessian(sycl::queue& q,
     return make_symmetric;
 }
 
+template <typename Float>
+sycl::event compute_raw_hessian(sycl::queue& q,
+                                const ndview<Float, 1>& probabilities,
+                                ndview<Float, 1>& out_hessian,
+                                const event_vector& deps) {
+    const std::int64_t n = probabilities.get_dimension(0);
+
+    ONEDAL_ASSERT(out_hessian.get_dimension(0) == n);
+    ONEDAL_ASSERT(probabilities.has_data());
+    ONEDAL_ASSERT(out_hessian.has_mutable_data());
+
+    const auto kernel = [=](const Float& val, Float*) -> Float {
+        constexpr Float one(1);
+        return val * (one - val);
+    };
+
+    return element_wise(q, kernel, probabilities, nullptr, out_hessian, deps);
+}
+
+template <typename Float>
+logloss_hessp<Float>::logloss_hessp(sycl::queue& q,
+                                    const ndview<Float, 2>& data,
+                                    const Float L2,
+                                    const bool fit_intercept)
+        : q_(q),
+          data_(data),
+          L2_{ L2 },
+          fit_intercept_{ fit_intercept },
+          n_{ data.get_dimension(0) },
+          p_{ data.get_dimension(1) } {
+    raw_hessian_ = ndarray<Float, 1>::empty(q_, { n_ });
+    buffer_ = ndarray<Float, 1>::empty(q_, { n_ });
+}
+
+template <typename Float>
+sycl::event logloss_hessp<Float>::set_raw_hessian(const ndview<Float, 1>& raw_hessian,
+                                                  const event_vector& deps) {
+    ONEDAL_ASSERT(raw_hessian.get_dimension(0) == n_);
+    return copy(q_, raw_hessian_, raw_hessian, deps);
+}
+
+template <typename Float>
+ndview<Float, 1>& logloss_hessp<Float>::get_raw_hessian() {
+    return raw_hessian_;
+}
+
+template <typename Float>
+sycl::event logloss_hessp<Float>::compute_with_fit_intercept(const ndview<Float, 1>& vec,
+                                                             ndview<Float, 1>& out,
+                                                             const event_vector& deps) {
+    auto* const buffer_ptr = buffer_.get_mutable_data();
+    const auto* const hess_ptr = raw_hessian_.get_data();
+    auto* const out_ptr = out.get_mutable_data();
+    ONEDAL_ASSERT(vec.get_dimension(0) == p_ + 1);
+    ONEDAL_ASSERT(out.get_dimension(0) == p_ + 1);
+    auto fill_buffer_event = fill<Float>(q_, buffer_, Float(1), deps);
+    auto out_suf = out.get_slice(1, p_ + 1);
+    auto out_bias = out.get_slice(0, 1);
+    auto vec_suf = vec.get_slice(1, p_ + 1);
+
+    sycl::event fill_out_event = copy(q_, out_suf, vec_suf, deps);
+    sycl::event fill_out0_event = fill<Float>(q_, out_bias, Float(0), deps);
```

Comment:
```suggestion
    sycl::event copy_event = copy(q_, out_suf, vec_suf, deps);
    sycl::event fill_out_event = fill<Float>(q_, out_bias, Float(0), deps);
```
Naming should be readable and describable